src/ngraph/op/fused/shuffle_channels.cpp

```diff
@@ -0,0 +1,93 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "ngraph/op/fused/shuffle_channels.hpp"
+#include "ngraph/op/reshape.hpp"
+#include "ngraph/op/util/reshape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+op::ShuffleChannels::ShuffleChannels(const shared_ptr<Node>& data, const int axis, const int groups)
+    : FusedOp("ShuffleChannels", {data})
+    , m_axis{axis}
+    , m_groups{groups}
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::ShuffleChannels::pre_validate_and_infer_types()
+{
+    NODE_VALIDATION_CHECK(this,
+                          m_axis >= 0,
```

Comment:
I've changed it and now the constructor can also accept negative values meaning counting from the back. Can't use size_t in this case since the validation happens in a separate method.